pin/pin.go

```diff
@@ -126,7 +127,7 @@ func (p *pinner) Pin(ctx context.Context, node *mdag.Node, recurse bool) error {
 func (p *pinner) Unpin(ctx context.Context, k key.Key, recursive bool) error {
 	p.lock.Lock()
 	defer p.lock.Unlock()
-	reason, pinned, err := p.isPinned(k)
```

Comment:
Maybe we should reuse the `PinMode` constants from line 31 or so. We would need to add `Indirect` and `Any` to the list, but it might be cleaner than using string constants (The string constants are my fault)